### PR TITLE
unit tests for elrond -> eth state machine

### PR DESCRIPTION
**Pull request by dragos-rebegea**
-added unit tests for state machine on Elrond to Ethereum flow
-added t.Parallel() for unit tests for state machine on Ethereum to Elrond flow

### CODE REVIEW

File: ethToElrond/v2/elrondToEth/steps/step01GetPending.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func (step *getPendingStep) Execute(ctx context.Context) (core.StepIdentifier, e
 	wasPerformed, err := step.bridge.WasTransferPerformedOnEthereum(ctx)
 	if err != nil {
 		step.bridge.GetLogger().Error("error determining if transfer was performed or not", "error", err)
-		return elrondToEth.GettingPendingBatchFromElrond, nil
+		return step.Identifier(), nil
```
Comment (iulianpascalau) [inline ethToElrond/v2/elrondToEth/steps/step01GetPending.go:40]: :+1:

File: ethToElrond/v2/elrondToEth/steps/step09WaitForQuorumOnSetStatus.go [hunk 2]
```diff
@@ -7,12 +7,12 @@ import (
 	"github.com/ElrondNetwork/elrond-eth-bridge/ethToElrond/v2/elrondToEth"
 )
 
-type waitForQuorumOnSetStatus struct {
+type waitForQuorumOnSetStatusStep struct {
```
Comment (iulianpascalau) [inline ethToElrond/v2/elrondToEth/steps/step09WaitForQuorumOnSetStatus.go:10]: :+1:

**Review by iulianpascalau — APPROVED** [review]
maybe a follow up PR with semi-integration tests?

File: ethToElrond/v2/elrondToEth/steps/step02SignProposedTransfer_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		assert.False(t, step.IsInterfaceNil())
+
+		expectedStepIdentifier := core.StepIdentifier(elrondToEth.WaitingForQuorumOnTransfer)
+		stepIdentifier, err := step.Execute(context.Background())
+		assert.Nil(t, err)
+		assert.Equal(t, expectedStepIdentifier, stepIdentifier)
+	})
+}
+
+func createStubeEecutorSignproposedtransfer() *bridgeV2.ElrondToEthBridgeStub {
```
Comment (sstanculeanu) [inline ethToElrond/v2/elrondToEth/steps/step02SignProposedTransfer_test.go:69]: typo:
```suggestion
func createStubExecutorSignProposedTransfer() *bridgeV2.ElrondToEthBridgeStub {
```

File: ethToElrond/v2/elrondToEth/steps/step05WaitTransferConfirmation_test.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/core"
+	"github.com/ElrondNetwork/elrond-eth-bridge/ethToElrond/v2/elrondToEth"
+	"github.com/ElrondNetwork/elrond-eth-bridge/testsCommon/bridgeV2"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExecute_WaitTrransferConfirmation(t *testing.T) {
```
Comment (sstanculeanu) [inline ethToElrond/v2/elrondToEth/steps/step05WaitTransferConfirmation_test.go:14]: typo:
```suggestion
func TestExecute_WaitTransferConfirmation(t *testing.T) {
```

File: ethToElrond/v2/elrondToEth/steps/step07ProposeSetStatus.go [hunk 5]
```diff
@@ -42,7 +41,7 @@ func (step *proposeSetStatusStep) Execute(ctx context.Context) (core.StepIdentif
 		return elrondToEth.GettingPendingBatchFromElrond, nil
 	}
 
-	return ethToElrond.SigningProposedTransferOnElrond, nil
+	return elrondToEth.SigningProposedSetStatusOnElrond, nil
```
Comment (sstanculeanu) [inline ethToElrond/v2/elrondToEth/steps/step07ProposeSetStatus.go:44]: :+1:

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**